openedx/core/djangoapps/programs/utils.py

```diff
@@ -142,6 +179,11 @@ def progress(self, programs=None, count_only=True):
             list of dict, each containing information about a user's progress
                 towards completing a program.
         """
+        now = datetime.datetime.now(utc)
+        enrolled_run_modes = defaultdict(set)
```

Comment:
There's a unique constraint on user and course (run) ID. So no, it's not possible to have more than one enrollment for a given course run.